marketplace/netlify-build/src/netlify-integration.js

```diff
@@ -0,0 +1,139 @@
+        const hook = { event, url };
+        return NetlifyClient.createNotificationHook(siteId, accessToken, hook);
+      });
+
+      return acc.concat([Promise.all(promisesForSite)]);
+    },
+    []
+  );
+
+  const netlifyHooks = await Promise.all(netlifyHookPromises);
+
+  // Merge flattened notification hook IDs to configuration.
+  updatedConfig.netlifyHookIds = netlifyHooks.reduce((acc, hooksForSite) => {
+    return acc.concat(hooksForSite.map(h => h.id));
+  }, []);
+
+  return updatedConfig;
+}
+
+export async function update({ config, accessToken }) {
+  config = prepareConfig(config);
+
+  // Remove existing hooks.
+  await removeExistingHooks({ config, accessToken });
+
+  // Remove references to removed hooks from configuration.
+  const updatedConfig = {
+    sites: config.sites.map(siteConfig => omit(siteConfig, ['buildHookId']))
+  };
+
+  // Proceed as in the installation step.
+  return install({ config: updatedConfig, accessToken });
+}
+
+function prepareConfig(config = {}) {
+  const sites = (config.sites || []).map(siteConfig => {
+    return { ...siteConfig, name: (siteConfig.name || '').trim() };
+  });
+
+  const netlifyHookIds = [...(config.netlifyHookIds || [])];
+
+  validateSiteConfigs(sites);
+
+  return { sites, netlifyHookIds };
+}
+
+function validateSiteConfigs(siteConfigs) {
+  // At least one site needs to be configured.
+  if (siteConfigs.length < 1) {
+    throw makeError('Provide at least one site configuration.');
+  }
+
+  // Find all site configurations with incomplete information.
+  const incomplete = siteConfigs.filter(siteConfig => {
+    return !siteConfig.netlifySiteId || siteConfig.name.length < 1;
+  });
+
+  if (incomplete.length > 0) {
+    throw makeError('Pick a Netlify site and provide a name for all configurations.');
+  }
+
+  // Display names must be unique.
+  const uniqueNames = uniqBy(siteConfigs, config => config.name);
+
+  if (uniqueNames.length !== siteConfigs.length) {
+    throw makeError('Display names must be unique.');
+  }
+}
+
+function makeError(message) {
+  const err = new Error(message);
+  err.useMessage = true;
+  return err;
+}
+
+async function removeExistingHooks({ config, accessToken }) {
+  const siteConfigs = config.sites || [];
+  const netlifyHookIds = config.netlifyHookIds || [];
+
+  // ...remove build hooks for it and...
+  const buildHookRemovalPromises = siteConfigs.map(siteConfig => {
+    const { netlifySiteId, buildHookId } = siteConfig;
+    if (netlifySiteId && buildHookId) {
+      return NetlifyClient.deleteBuildHook(netlifySiteId, buildHookId, accessToken);
+    } else {
+      return Promise.resolve();
+    }
+  });
+
+  // ...remove Netlify hooks for it.
+  let netlifyHookRemovalPromises = Promise.resolve();
+
+  if (Array.isArray(netlifyHookIds)) {
+    const validNetlifyHookIds = netlifyHookIds.filter(id => {
+      return typeof id === 'string' && id.length > 0;
+    });
+
+    netlifyHookRemovalPromises = Promise.all(
```

Comment:
Minor: I don't think it's necessary to do a `Promise.all` here since it concats with the array on line 134. It might be better to remove it here since the logic below handles it in a `try...catch` where this one will throw.